src/parser/FileParser.kt

```diff
@@ -0,0 +1,108 @@
+package parser
+
+import CoffeeShop
+import Coordinate
+import java.io.File
+
+/**
+ * Defines a parser.FileParser object, containing different operations on the content of a given .csv file
+ *
+ * created by AlexandraV on 12/25/2018
+ */
+data class FileParser(var fileName: String, var userCoordinate: Coordinate) {
+    val coffeeShopsList: MutableList<CoffeeShop> = mutableListOf()
+    private val lines: List<String>? = File(fileName).useLines { it.toList() } // the reader is closed automatically
+
+    /**
+     *  parses the lines of the .csv file and converts them into CoffeeShops objects, if possible
+     *  All the lines that cannot be converted to CoffeeShops objects will be written into  invalid_coffee_shops.txt file
+     */
+    fun parseCoffeeShopsList(n: Int) {
+        val invalidCoffeeShopsList: MutableList<String> = mutableListOf()
+        if (lines != null) {
+            var lineNumber = 1
+            lines.forEach { it ->
+                val result: List<String> = it.split(",").map { it.trim() }
+                if (isCSVLineValid(result)) {
+                    insertShopIntoList(result[0], result[1], result[2], n)
+                } else {
+                    if (!it.trim().isEmpty()) invalidCoffeeShopsList.add("Line $lineNumber: $it \n")
+                }
+                lineNumber++
+            }
+
+            if (n <= coffeeShopsList.size) {
+                coffeeShopsList.subList(0, n).forEach {
+                    println("${it.shopName}, ${it.shopDistanceToUser}")
+                }
+            } else {
+                if (coffeeShopsList.size > 0) {
+                    coffeeShopsList.forEach { println("${it.shopName}, ${it.shopDistanceToUser}") }
+                    println("\nThe number of CoffeeShops you requested ($n) exceeds the number of valid shops in the .csv file. Only ${coffeeShopsList.size} result(s) printed")
+                } else {
+                    println("There are no valid coffee shops in $fileName to be displayed")
+                }
+            }
+        }
+        // write all invalid lines (if there are any) in a file
+        if (!invalidCoffeeShopsList.isEmpty()) {
+            println("\nAll invalid lines found in $fileName are written in invalid_coffee_shops.txt")
+            File("invalid_coffee_shops.txt").bufferedWriter().use { out ->
+                invalidCoffeeShopsList.forEach {
+                    out.write(it)
+                }
+            }
+        }
```

Comment:
This method is quite hard to follow and does more things that it should. One shouldn't need comments to explain what the code does.